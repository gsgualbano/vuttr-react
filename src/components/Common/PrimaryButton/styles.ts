import styled from 'styled-components';
import { Theme } from '~/styles/global';

interface ButtonProps {
  outline: boolean;
}

type Props = ButtonProps & Theme;

export const Button = styled.button`
  display: inline-flex;
  flex-direction: row;
  font-size: 16px;
  color: ${({ outline }: ButtonProps) => (outline ? '#365df0' : '#ffffff')};
  background: ${({ outline, theme }: Props) => (outline ? theme.colors.buttonOutlineBackground : '#365df0')};
  padding: 5px 10px;
  border-radius: 5px;
  border: none;
  justify-content: center;
  align-items: center;

  &:hover {
    ${({ outline }: ButtonProps) => (outline ? 'color: #2f55cc' : 'background: #2f55cc')};
  }

  &:click {
    ${({ outline }: ButtonProps) => (outline ? 'color: #244aa8' : 'background: #244aa8')};
    background-color: #244aa8;
  }

  @media (max-width: 768px) {
    width: 100%;
  }
`;
