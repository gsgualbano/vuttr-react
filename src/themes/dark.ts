export default {
  colors: {
    background: '#1B1A1A',
    textColor: '#fff',
    cardBackground: '#29292B',
    cardBorderColor: '#29292B',
    modalBackground: '#29292B',
    modalBorderColor: '#29292B',
    linkColor: '#ae8fd6',
    linkVisitedColor: '##8050bf',
    buttonOutlineBackground: '#29292B',
    InputBackground: '#9e9fa2',
    InputBorderColor: '#9e9fa2',
  },
};
