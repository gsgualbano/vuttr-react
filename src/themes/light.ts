export default {
  colors: {
    background: '#fcfcfd',
    textColor: '#170c3a',
    cardBackground: '#fcfcfd',
    cardBorderColor: '#ebeaed',
    modalBackground: '#fefefe',
    modalBorderColor: '#888',
    linkColor: 'rgb(0, 0, 238)',
    linkVisitedColor: 'rgb(85, 26, 139);',
    buttonOutlineBackground: '#ffffff',
    InputBackground: '#f5f4f6',
    InputBorderColor: '#ebeaed',
  },
};
