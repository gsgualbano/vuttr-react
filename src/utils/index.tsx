/* eslint-disable max-len */
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import { render } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import { reducer } from '../store/ducks/tools';

interface Theme {
  colors: {
    background: string;
    textColor: string;
    cardBackground: string;
    cardBorderColor: string;
    modalBackground: string;
    modalBorderColor: string;
    linkColor: string;
    linkVisitedColor: string;
    buttonOutlineBackground: string;
    InputBackground: string;
    InputBorderColor: string;
  };
}

/** redux mock to render components that depends on the redux */
export const renderWithRedux = (
  component: JSX.Element,
  { initialState, store = createStore(reducer, initialState) }: any = {},
) => ({
  ...render(<Provider store={store}>{component}</Provider>),
  store,
});

/** render components thats depends on the theme provider */
export const renderWithTheme = (component: JSX.Element, theme: Theme) => render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

/** render components thats depends on the redux and theme provider */
export const renderWithReduxAndTheme = (
  component: JSX.Element,
  theme: Theme,
  { initialState, store = createStore(reducer, initialState) }: any = {},
) => ({
  ...render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>{component}</Provider>
    </ThemeProvider>,
  ),
  store,
});
