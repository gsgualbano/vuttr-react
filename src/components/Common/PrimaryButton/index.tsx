import React from 'react';
import { Button } from './styles';

/**
 * Props do PrimaryButton.
 */
interface OwnProps {
  /** elemento jsx */
  children?: React.ReactNode;
  /** uma ação ser executa após o click no botão como um console.log ou função  */
  onClick?(e: React.MouseEvent<HTMLElement, MouseEvent>): void;
  /** inverter as cores do botão */
  outline?: boolean;
  /** tipo do botão */
  type?: 'submit' | 'reset' | 'button';
}

/**
 * Botão primário
 */
const PrimaryButton = ({
  type = 'button',
  outline = false,
  children,
  onClick,
}: OwnProps): JSX.Element => (
  <Button type={type} onClick={onClick} outline={outline} data-testid="primary-button">
    {children}
  </Button>
);

export default PrimaryButton;
