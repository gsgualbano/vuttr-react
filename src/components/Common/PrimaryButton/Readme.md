#### Botão com a cor primária(Azul).

Regular PrimaryButton Example:

```js
<PrimaryButton>Example</PrimaryButton>
```

Outline DangerButton Example:

```js
import { ThemeProvider } from "styled-components";
import light from "~/themes/light";

<ThemeProvider theme={light}>
  <PrimaryButton outline>Example</PrimaryButton>
</ThemeProvider>;
```

Dark Outline DangerButton Example:

```js
import { ThemeProvider } from "styled-components";
import dark from "~/themes/dark";

<ThemeProvider theme={dark}>
  <PrimaryButton outline>Example</PrimaryButton>
</ThemeProvider>;
```
