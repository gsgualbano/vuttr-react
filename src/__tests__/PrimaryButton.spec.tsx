import React from 'react';
import { cleanup } from '@testing-library/react';
import { renderWithTheme } from '~/utils';

import PrimaryButton from '~/components/Common/PrimaryButton';

import 'jest-dom/extend-expect';
import 'jest-styled-components';

import dark from '~/themes/dark';
import light from '~/themes/light';

describe('<PrimaryButton />', () => {
  afterEach(cleanup);

  it('render correctly', () => {
    const { asFragment } = renderWithTheme(<PrimaryButton>Add</PrimaryButton>, light);
    expect(asFragment).toMatchSnapshot();
  });

  it('have correct color', () => {
    const { getByTestId } = renderWithTheme(<PrimaryButton>Add</PrimaryButton>, light);
    expect(getByTestId('primary-button')).toHaveStyleRule('background', '#365df0');
    expect(getByTestId('primary-button')).toHaveStyleRule('color', '#ffffff');
    expect(getByTestId('primary-button')).toHaveStyleRule('background', '#2f55cc', {
      modifier: ':hover',
    });
  });

  it('have correct outline color', () => {
    const { getByTestId } = renderWithTheme(<PrimaryButton outline>Add</PrimaryButton>, light);
    expect(getByTestId('primary-button')).toHaveStyleRule('background', '#ffffff');
    expect(getByTestId('primary-button')).toHaveStyleRule('color', '#365df0');
    expect(getByTestId('primary-button')).toHaveStyleRule('color', '#2f55cc', {
      modifier: ':hover',
    });
  });

  it('have correct dark outline color', () => {
    const { getByTestId } = renderWithTheme(<PrimaryButton outline>Add</PrimaryButton>, dark);
    expect(getByTestId('primary-button')).toHaveStyleRule('background', '#29292B');
    expect(getByTestId('primary-button')).toHaveStyleRule('color', '#365df0');
    expect(getByTestId('primary-button')).toHaveStyleRule('color', '#2f55cc', {
      modifier: ':hover',
    });
  });
});
