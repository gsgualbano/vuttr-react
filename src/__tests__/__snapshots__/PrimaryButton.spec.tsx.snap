// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> render correctly 1`] = `[Function]`;
